### PR TITLE
feat(pedersen-hash): add None personalization

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

File: src/circuit/pedersen_hash.rs [hunk 1]
```diff
@@ -25,10 +25,12 @@ where
     CS: ConstraintSystem<E>,
 {
     let personalization = personalization.get_constant_bools();
-    assert_eq!(personalization.len(), 6);
 
     let mut edwards_result = None;
     let mut bits = personalization.iter().chain(bits.iter());
+    let k = bits.clone();
+    println!("{:?}", k.collect::<Vec<_>>().len());
```
Comment (porcuquine) [inline src/circuit/pedersen_hash.rs]: I assume you meant to remove this.

**Review by porcuquine — APPROVED** [review]
Hooray. Just needs stray `println` removed.

**Comment by nicola** [comment]
Can you report the saving that you have with this?

Why can't we just "cache" the personalization for all pedersen?, that should not add any constraints (except a small constant)

**Comment by dignifiedquire** [comment]
> Why can't we just "cache"

personalizations are dependent on the height in the merkle tree, so I don't understand how we could cache this in the circuit

**Comment by dignifiedquire** [comment]
> Can you report the saving that you have with this?

You can see the circuit size changes in the diff here: https://github.com/filecoin-project/rust-fil-proofs/pull/759

**Comment by nicola** [comment]
ok, it does look that the saving for pedersen is very minimal and it looks like that some form of caching is kicking in (not for blake2s)

**Pull request merged**